ADDONS.md

```diff
@@ -219,33 +219,40 @@ class MyFormatterRubyLspAddon < RubyLsp::Addon
     "My Formatter"
   end
 
-  def activate(message_queue)
+  def activate(global_state, message_queue)
     # The first argument is an identifier users can pick to select this formatter. To use this formatter, users must
     # have rubyLsp.formatter configured to "my_formatter"
     # The second argument is a singleton instance that implements the `FormatterRunner` interface (see below)
-    RubyLsp::Requests::Formatting.register_formatter("my_formatter", MyFormatterRunner.instance)
+    global_state.register_formatter("my_formatter", MyFormatterRunner.instance)
   end
 end
 
-# Custom formatting runner
-class MyFormatterRunner
+# Custom formatter
+class MyFormatter
   # Make it a singleton class
   include Singleton
   # If using Sorbet to develop the addon, then include this interface to make sure the class is properly implemented
-  include RubyLsp::Requests::Support::FormatterRunner
+  include RubyLsp::Requests::Support::Formatter
 
   # Use the initialize method to perform any sort of ahead of time work. For example, reading configurations for your
   # formatter since they are unlikely to change between requests
   def initialize
+    @config = read_config_file!
   end
 
-  # The main part of the interface is implementing the run method. It receives the URI and the document being formatted.
-  # IMPORTANT: This method must return the formatted document source without mutating the original one in document
-  def run(uri, document)
+  # IMPORTANT: None of the following methods should mutate the document in any way or that will lead to a corrupt state!
+
+  # Provide formatting for a given document. This method should return the formatted string for the entire document
+  def formatting(uri, document)
```

Comment:
I was thinking that too, but then would `diagnostic` then become `diagnose`? That felt wrong.

What if we prefix then to make them into 'verb phrases', e.g. `run_formatting` and `run_diagnostic`?

Reply:
I think the LSP spec is different because of the `document/` prefix. So, in that case, the client is making a request to do "document formatting", or a request to perform "document diagnostic" (and that's why it is singular).

Since our usage is more like `@active_formatter.formatting(@uri, @document)`, I think it reads wrong. It would read much nicer if it was `@active_formatter.format(@uri, @document)`. Similarly, we could have `@active_formatter.diagnose(@uri, @document)` but, imo,  `diagnostic` works as is for that case.

Reply:
I also prefer verbs here. IMO readability in the Ruby context (verbs for methods) is more important than matching the terms used in specification.

Reply:
I just tried and calling the method `format` throws Sorbet off because it thinks we're referring to `Kernel#format`.

```
lib/ruby_lsp/requests/support/syntax_tree_formatter.rb:36: Return type T.nilable(String) does not match return type of overridden method Kernel#format https://srb.help/5035
    36 |        def format(uri, document)
                ^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/7058a6b736dde975d7759f2e43cdf805970ae6af/rbi/core/kernel.rbi#L1585: Super method defined here with return type String
    1585 |  def format(format, *args); end
```

Should we go with Andy's suggestion and use `run_formatting`?

Reply:
Switch to `run_formatting` and `run_diagnostic`.